internal/store/pod.go

```diff
@@ -769,6 +769,110 @@ var (
 				}
 			}),
 		},
+		{
+			Name: "kube_pod_container_resource_requests_cpu_cores",
+			Type: metric.Gauge,
+			Help: "The number of CPU cores requested by a container.",
+			GenerateFunc: wrapPodFunc(func(p *v1.Pod) *metric.Family {
+				ms := []*metric.Metric{}
+
+				for _, c := range p.Spec.Containers {
+					req := c.Resources.Requests
+
+					for resourceName, val := range req {
+						if resourceName == v1.ResourceCPU {
+							ms = append(ms, &metric.Metric{
+								LabelKeys:   []string{"container"},
+								LabelValues: []string{c.Name},
+								Value:       float64(val.MilliValue()) / 1000,
```

Comment:
Why not just use the ScaledValue with the wanted Scale? IIRC with MilliValue we could potentially lose some data if its not the thing we want in the end. https://github.com/kubernetes/apimachinery/blob/master/pkg/api/resource/amount.go#L36-L46

Reply:
This is what we've always been doing with CPU resources, and have never had a problem with. My understanding is that CPU request always gets parsed in millis, so that's why there is no precision loss in this case. I agree the API of these types is very confusing, but this appears to be correct.

Reply:
Not judging by the code https://github.com/kubernetes/apimachinery/blob/master/pkg/api/resource/quantity.go#L702

Reply:
I'm fine with revisiting this, just this has worked for ~3 years, I'd be hesitant to mix these two changes.

Reply:
Sounds good to me! Lets revisit if we get problems then :)